src/wp-includes/media.php

```diff
@@ -1723,25 +1736,57 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 			$value = 'lazy';
 		}
 
-		// Images should have dimension attributes for the `loading` attribute to be added.
-		if ( false === strpos( $image, ' width=' ) || false === strpos( $image, ' height=' ) ) {
+		// Images should have source and dimension attributes for the `loading` attribute to be added.
+		if ( false === strpos( $image, ' src=' ) || false === strpos( $image, ' width=' ) || false === strpos( $image, ' height=' ) ) {
 			return $image;
 		}
 
-		$quote = null;
+		return str_replace( '<img', '<img loading="' . esc_attr( $value ) . '"', $image );
```

Comment:
```suggestion
		// The $value is hard-coded to either lazy or eager. See `in_array()` above.
		return str_replace( '<img', '<img loading="' . $value . '"', $image );
```

Thinking `esc_attr()` is not needed for hard-coded strings? Can only be `lazy` or `eager`, see `in_array()` above. A little speed-up?

Reply:
Hmm, better safe than sorry here I think. Since this is a string, I think we're better off escaping.